### ISSUE TITLE
Factorial of a negative number does not exist

### ISSUE THREAD

**Issue by njiddasalifu** [trekhleb/javascript-algorithms#1008]
This issue is addressing the fact that in the factorialrecursive implementation, it says if number is not greater than 1 then return 1. its not true for all cases cox fact of a neative is undefined.

**Comment by njiddasalifu** [comment]
I have solved this already check PR for merged

**Comment by HemanthPaila** [comment]
Hi there,

I want to contribute to this project and take on this issue as my first task. Could you please assign it to me? I am excited to work towards resolving it and will keep the team updated on my progress.

Thanks!

**Comment by xyn22** [comment]
> I have solved this already check PR for merged

Please link your PR to this issue.

**Comment by njiddasalifu** [comment]
@xyn22  you can find the PR link here https://github.com/trekhleb/javascript-algorithms/pull/1021

**Comment by Leonardbrett** [comment]
> Hi there,
> 
> I want to contribute to this project and take on this issue as my first task. Could you please assign it to me? I am excited to work towards resolving it and will keep the team updated on my progress.
> 
> Thanks!

hello everyone I'm super excited I can't stop recommending her she help me a lot on trading I just made my withdraw of 22,108$ from my start of 5,800$ from Charlotte trading platform reach her on WhatsApp:+𝟭𝟰𝟳𝟱𝟮𝟰𝟮𝟲𝟯𝟯𝟴

**Comment by lazarljubenovic** [comment]
But does she know what's the factorial of 0?

**Comment by cynthiamarshal** [comment]
This issue pertains to the factorial recursive implementation. Currently, the implementation states that if the number is not greater than 1, it should return 1. However, this is not entirely accurate because the factorial of a negative number is undefined. [DGme](https://dgmeportal.live/)

**Comment by utkarsh-shrivastav77** [comment]
Hi is this issue still open

**Comment by jesicaakon** [comment]
That is correct. The factorial of a negative number does not exist. The factorial function is defined as the product of all positive integers less than or equal to a given number. Since negative numbers are not positive, they cannot be part of the product. Therefore, the factorial of a negative number is undefined.

Here are some examples of factorials of positive integers:

1! = 1
2! = 2
3! = 6
4! = 24
5! = 120
Here are some examples of factorials of negative integers:

(-1)! = undefined
(-2)! = undefined
(-3)! = undefined
(-4)! = undefined
(-5)! = undefined
You can also check factorial types of issues at [Employee Connection](https://ouremployeeconnection.com/)

**Comment by resicajasi** [comment]
the factorial recursive implementation, the statement that returns 1 when the number is not greater than 1 is not suitable for all cases. This is because the factorial of a negative number is undefined. It's important to handle negative numbers as a special case and either return an error or provide appropriate validation to prevent undefined behavior in the factorial function. For further you can look more deep down on through website [getmyccpay.com](https://getmyccpay.com/)

**Comment by lazarljubenovic** [comment]
The GPT spam era is starting ☺️

**Comment by Anekantjainsagar** [comment]
Facing an issue after pulling code from GitHub while running **npm run lint**

**Comment by ZepLeonard** [comment]
您的信件已经收到，我会尽快查阅，谢谢！ ——林泽鹏
This is an automatic reply, confirming that your e-mail was received.Thank you

**Comment by raidenkhan** [comment]
> the factorial recursive implementation, the statement that returns 1 when the number is not greater than 1 is not suitable for all cases. This is because the factorial of a negative number is undefined. It's important to handle negative numbers as a special case and either return an error or provide appropriate validation to prevent undefined behavior in the factorial function. I would invite you to visit my blog for more informative updates [getmyccpay.com](https://getmyccpay.com/)

Yes I agree the base case in the recurssive statement should state if number <=1 return 1 which handles pretty much everythin

**Comment by ihemanthm** [comment]
I agree with the person stated that the factorial of number <0 i.e, -1!,  -2!,... all results undefined whereas the values of -0!,0! and 1! results as 1 and the rest continues

**Comment by sumukhj1219** [comment]
Assign it to me

**Comment by Sanil-Surve** [comment]
Hii I am interested in solving this issue

**Comment by njiddasalifu** [comment]
This has been solved already.
